contracts/cw20-ics20/src/ibc.rs

```diff
@@ -324,9 +325,6 @@ fn on_packet_success(deps: DepsMut, packet: IbcPacket) -> Result<IbcBasicRespons
         attr("success", "true"),
     ];
 
-    // we have made a proper transfer, record this
```

Comment:
It will be nice to have something like a double check here. If this arrives, we confirm the balance adjustment. If this "never" arrives, we reject it.

This would require something like a queue / pool of pending balance adjustment confirmations... also, what happens if the deposit is successful in the other side, but the success packet never arrives. Does IBC has something like a "guaranteed delivery" mechanism for these packets?

Reply:
There is no guarantee they ever arrive if the relayers are down, but an ack will remain waiting on the remote queue until it is handled.